### DISCUSSION TITLE
Feature: Checkout process

### DISCUSSION THREAD

**Discussion by JuanDelPueblo** [uprm-inso4101-2024-2025-s1/semester-project-mercado-colegial-online#3] [Ideas]
A key aspect of any online marketplace is the checkout process where customers can finish the purchase of one or more items by entering their personal information, shipping address, and payment method. This feature will need to be integrated into this project for users to be able to purchase items instantly without having to contact the seller directly. It can be triggered directly from a buy button attached to the item, or a shopping cart system can be added where multiple items can be purchased all at once.

**Comment by JoshDG03** [comment]
Diving a little bit into specifics, having some type of "ath movil" redirect that fills in the amount to pay automatically when you want your order to be processed and then sends the order over to the seller once the payment has been received would seem ideal for this checkout process, since these student owned businesses typically already rely on "ath movil" for transactions.

**Comment by CdoxHype** [comment]
We can have the seller register different ways they would like to be paid, such as PayPal, Stripe, or things like that, so that the buyer has those different ways to pay as well.These methods would avoid the payment confirmation problem that ATH Mobile has. For this, the merchant must know which payment corresponds to each person. If ATH Mobile is used as a payment method, a way must be thought of that our website can confirm that that person paid.

**Reply by JuanDelPueblo** [reply]
There does exist an API for ATH Movil that allows you to pay a set amount automatically but it is dependent on the use of a business account, which not every seller might have. See https://github.com/evertec/athmovil-javascript-api for more details

**Comment by david-baez-bravo** [comment]
An option to pick up the product should be added. Or an option for meeting at the university would also be convenient. Shipping products and paying or charging for shipping is time consuming and unprofitable for many sellers.

**Comment by JuanIranzo** [comment]
I looked a bit into how the PayPal API works and it also has a way of parallel payment, meaning that if the buyer has multiple products in the cart, but not all of them are from the same seller, when they pay through PayPal, it splits the total payment into the corresponding sub-payments for the different sellers. Additionally, from what I saw, the PayPal API has a way to test if purchases are going through as they should, without the need of real money, which is something really important in our case.
